Bulk Actions on Pipeline List

camielvs opened the pull request:
## Description

<!-- Please provide a brief description of the changes made in this pull request. Include any relevant context or reasoning for the changes. -->
Adds checkboxes and bulk actions to the pipeline list. Users can now select multiple pipelines and delete them.

Bulk actions are access by an action bar that appears at the bottom of the screen when some checkboxes have been selected


## Related Issue and Pull requests

<!-- Link to any related issues using the format #<issue-number> -->

User requested

## Type of Change

<!-- Please delete options that are not relevant -->

- [ ] Bug fix
- [x] New feature
- [x] Improvement
- [ ] Cleanup/Refactor
- [ ] Breaking change
- [ ] Documentation update

## Checklist

<!-- Please ensure the following are completed before submitting the PR -->

- [ ] I have tested this does not break current pipelines / runs functionality
- [ ] I have tested the changes on staging

## Screenshots (if applicable)

<!-- Include any screenshots that might help explain the changes or provide visual context -->
<img width="1581" height="1259" alt="image" src="https://github.com/user-attachments/assets/727a4eab-e407-4a54-a762-979480607abf" />



## Test Instructions

<!-- Detail steps and prerequisites for testing the changes in this PR -->
1. Navigate to the pipeline list
2. select some pipelines using the checkboxes
4. delete the pipelines
5. (de)select all using the checkbox at the top of the table

## Additional Comments

<!-- Add any additional context or information that reviewers might need to know regarding this PR -->

maxy-shpfy commented on src/components/Home/PipelineSection/PipelineSection.tsx:
```diff
@@ -86,6 +91,34 @@ export const PipelineSection = () => {
 
+  const handleSelectAll = (checked: boolean) => {
+    if (checked) {
+      const allPipelineNames = new Set(filteredPipelines.map(([name]) => name));
```
not a blocker. not ready to offer something better, but I'm not enjoying `Set` manipulation via creating new objects all the time.

camielvs replied:
I don't have a good answer to this at the moment. I asked AI and they didn't have anything useful to say at this point either 🤷‍♂️ let's leave it for future

maxy-shpfy commented on src/components/Home/PipelineSection/BulkActionsBar.tsx:
```diff
@@ -0,0 +1,110 @@
+      await Promise.all(deletePromises);
+      onDelete();
+    } catch (error) {
+      console.error("Failed to delete some pipelines:", error);
```
I would prefer an explicit message to user via a toast notification.

camielvs replied:
Done!

maxy-shpfy commented on src/components/Home/PipelineSection/BulkActionsBar.tsx:
```diff
@@ -0,0 +1,110 @@
+  return (
+    <div className="fixed bottom-4 left-1/2 transform -translate-x-1/2 bg-background border border-border rounded-lg shadow-lg p-4 z-50">
+      <div className="flex items-center gap-4">
+        <span className="text-sm font-medium">
```
NIT: using UI primitives

maxy-shpfy commented on src/components/Home/PipelineSection/BulkActionsBar.tsx:
```diff
@@ -0,0 +1,110 @@
+
+      const newWindow = window.open(url, uniqueTabId);
+
+      if (newWindow === null || typeof newWindow === "undefined") {
+        console.error(`Tab ${uniqueTabName} was blocked by popup blocker`);
+        alert(
+          `Tab ${uniqueTabName} and subsequent tabs were blocked by your browser's popup blocker. ${i} tab(s) opened successfully. Please allow popups for Oasis to open all tabs at once.`,
+        );
+        break;
```
I feel this is a bad UX. It is not obvious on HOW to allow any popups from the domain.  I would vote to not include Bulk opening at least unless we have a great evidence the feature it highly requested and there no workarounds.

maxy-shpfy commented on src/components/Home/PipelineSection/BulkActionsBar.tsx:
```diff
@@ -0,0 +1,110 @@
+      }
+
+      if (i < selectedPipelines.length - 1) {
+        await new Promise((resolve) => setTimeout(resolve, 200));
```
I dont like this approach. While I cant offer a better way, overall "waiting split" via microtasks is not the greatest wait to do it. 

I would think about using a queue of tabs to be opened + setInternval/setTimeout - those will schedule a next tick, versus abusing microtask queue.

camielvs replied:
With the removal of bulk-open this code has been removed.

maxy-shpfy reviewed (commented):
From my testing:

1.  I cant open more than one tab at once - it's got blocked by the browser. Also just a thought - I barely can imagine the need to open all Pipelines at once. But I may find useful ability to bulk-export pipelines (as separate files or as one archive - open question).

<img width="455" height="201" alt="image" src="https://github.com/user-attachments/assets/2a01d75b-2b96-42cf-97fd-01ac4f47aff6" />

What about instead add a button "open in new window" so users can open pipelines in new windows one by one. 

2. Minor UX: Since the entire row is clickable - some times even a small miss-click around the checkbox just opens it in this tab.

camielvs commented:
Updated -- I extracted a few common string items into new methods. Probably should be it's own PR, but it's it's here now. I can split out if desired.
